sklearn/pipeline.py

```diff
@@ -135,40 +222,53 @@ def _final_estimator(self):
 
     # Estimator interface
 
-    def _pre_transform(self, X, y=None, **fit_params):
-        fit_params_steps = dict((step, {}) for step, _ in self.steps)
+    def _fit(self, X, y=None, **fit_params):
+        fit_params_steps = dict((name, {}) for name, step in self.steps
+                                if step is not None)
         for pname, pval in six.iteritems(fit_params):
             step, param = pname.split('__', 1)
             fit_params_steps[step][param] = pval
         Xt = X
         for name, transform in self.steps[:-1]:
-            if hasattr(transform, "fit_transform"):
+            if transform is None:
+                pass
+            elif hasattr(transform, "fit_transform"):
                 Xt = transform.fit_transform(Xt, y, **fit_params_steps[name])
             else:
                 Xt = transform.fit(Xt, y, **fit_params_steps[name]) \
                               .transform(Xt)
         return Xt, fit_params_steps[self.steps[-1][0]]
 
     def fit(self, X, y=None, **fit_params):
-        """Fit all the transforms one after the other and transform the
+        """Fit the model
+
+        Fit all the transforms one after the other and transform the
         data, then fit the transformed data using the final estimator.
 
         Parameters
         ----------
         X : iterable
             Training data. Must fulfill input requirements of first step of the
             pipeline.
+
         y : iterable, default=None
             Training targets. Must fulfill label requirements for all steps of
             the pipeline.
+
+        fit_params : dict of string -> object
```

Comment:
this looks like ``fit_params`` is a named argument, but it's actually ``**kwargs``, right?